### ISSUE TITLE
Edge Quality Rating Evaluation

### ISSUE THREAD

**Issue by theodorelu27** [EdgeLab-FHDO/Edge-Diagnostic-Platform#98]
Implement Edge Quality Rating for the Client's Connection Evaluator

Components to be added to the Clients:

**Operator for EQR calculation**
- Counter for current EQR score
- Configureable Latency Treshold and score reduction
- Include it as information to be sent on the Latency Reporter ( #97 )

**Comment by theodorelu27** [comment]
@alyhasansakr I want to confirm the tasks for this issue before moving on, as I noticed there were some that I actually missed in the initial issue

Also, I want to confirm some things regarding the EQR:
- From what I understood in the design from your Thesis, EQR is assigned from the Master when a node is assigned to the Client, as such Client would receive a Max_EQR and sets its current_EQR to half of the maximum value
- EQR would decrease by a fixed amount `x`, when the latency passes latency_treshold `z`
- EQR would increase by a fixed amount `y` when a process is completed with latency bellow latency_treshold `z`
- As with the current system, the Client would still prioritize using Server when it is connected but the expectation is that when max_EQR is reached, only Edge Execution would be used
- EQR would be sent to matchmaker to evaluate connection history
- EQR value would be maintained and sent along with the latency report
- Client would maintain its own `connected` status but Server would maintain it based on input from Master
- When EQR reaches 0, matchmaker would assign a new Server node for the Client
- When EQR reaches 0, Client would disconnect (remove all information about Server and continue subsequent operation locally) and asks to be assigned a new node
- Value `x, y, z`  should be configurable when starting the Client

**Comment by alyhasansakr** [comment]
@theodorelu27 yes, that is correct.

**Comment by theodorelu27** [comment]
@alyhasansakr got it, but as for the EQR, as currently, the value is not yet available, would I need to implement it myself in the matchmaker or should I just prepare the interface for now and use a default value when the EQR was not provided by the Master?

**Comment by alyhasansakr** [comment]
@theodorelu27 No, for now just use a default value, we will implement it later.

**Comment by theodorelu27** [comment]
Additions needed to be made
- add an "Evaluating" state
- separate runner for remote and local execution
- add Max_EQR as a parameter

**Comment by theodorelu27** [comment]
@alyhasansakr for the evaluation, since we're measuring end-to-end latency, in the case of the marker detection, should I just draw the detected marker into a new file with a different name? (such as by adding "eval_" into the file name) or should I just skip marker drawing for evaluation?

**Comment by alyhasansakr** [comment]
@theodorelu27 just skip.
